### PR TITLE
MINOR: Change KStreamKstreamOuterJoinTest to use distinct left and right types

### PR DESCRIPTION
**Pull request by gharris1727**
This test uses the same value types on the left and right, and so wouldn't be sensitive to a mixup between left and right values. So I changed one of the stream types to `Long`, and updated the assertions to match.

Because the implementation of the KStreamKstreamJoin is not type-safe, it was unclear just from the code that a mixup was not possible.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamKStreamOuterJoinTest.java [hunk 1]
```diff
@@ -737,12 +739,12 @@ public void runOuterJoin(final StreamJoined<Integer, String, String> streamJoine
...
-                new KeyValueTimestamp<>(0, "C0+a0", 0L),
-                new KeyValueTimestamp<>(0, "C0+b0", 0L),
-                new KeyValueTimestamp<>(1, "C1+a1", 0L),
-                new KeyValueTimestamp<>(1, "C1+b1", 0L),
-                new KeyValueTimestamp<>(2, "C2+b2", 0L),
-                new KeyValueTimestamp<>(3, "C3+b3", 0L)
+                new KeyValueTimestamp<>(0, "C0+0", 0L),
+                new KeyValueTimestamp<>(0, "C0+0", 0L),
+                new KeyValueTimestamp<>(1, "C1+1", 0L),
+                new KeyValueTimestamp<>(1, "C1+1", 0L),
```
Comment (cadonna) [inline streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamKStreamOuterJoinTest.java]: This seems to make the verification less strict. It does not differentiate between `a0` and `b0` as well as `a1` and `b1`. Maybe use `10` for `a0` and `20` for `b0` and `11` for `a1` and `21` for `b1`.

**Review by cadonna — COMMENTED** [review]
Thanks for the change, @gharris1727 !

I have one comment.

File: streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamKStreamOuterJoinTest.java (hunk 1 above)
Comment (gharris1727) [reply streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamKStreamOuterJoinTest.java]: You're right, I didn't notice this. I did a search-and-replace renaming, and reverted the stuff which didn't make sense.

I did have to manually renumber stuff like "a0-0", and some places where capital letters "A0" were used on the inputStream2 to fit the pattern better. PTAL, thanks!

**Comment by mjsax** [comment]
@gharris1727 @cadonna -- do we still want to merge this PR?

**Comment by cadonna** [comment]
Yes, why not? It is a good change to the test, no?
I just did not have time to review it since my last review.

**Comment by mjsax** [comment]
Was just wondering if the other PR contained tests that cover this already.

@gharris1727 could you rebase so a new PR build is triggered to make sure everything still works? Happy to reveiw/merge afterwards.

**Comment by gharris1727** [comment]
@mjsax It was already up-to-date without merge conflicts, but I synced it up again.

**Comment by mjsax** [comment]
Thanks @gharris1727 -- merged to `trunk`.

**Pull request merged**